### PR TITLE
Docs update for Job Backoff Limit Per Index in Beta

### PR DESCRIPTION
**Pull request by mimowo**
Tracking issue: https://github.com/kubernetes/enhancements/issues/3850

### CODE REVIEW

**Comment by katcosgrove** [comment]
Hi, @mimowo! v1.29 Docs Lead here. Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday 14th November 2023. Thank you!

**Comment by mimowo** [comment]
/assign @alculquicondor @sftim 
Minimal changes for now, but there are no changes / bugfixes since 1.29. We could consider adding a tasks page as planned before for alpha, but would like to hear opinions of others before doing.

**Comment by alculquicondor** [comment]
I don't think it's worth adding a task page just for this feature.

/lgtm

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -115,6 +115,7 @@ For a reference to old feature gates that are removed, please refer to
 | `InTreePluginPortworxUnregister` | `false` | Alpha | 1.23 | |
 | `InTreePluginvSphereUnregister` | `false` | Alpha | 1.21 | |
 | `JobBackoffLimitPerIndex` | `false` | Alpha | 1.28 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:117]:
```suggestion
| `JobBackoffLimitPerIndex` | `false` | Alpha | 1.28 | 1.28 |
```

**Comment by tengqm** [comment]
Thanks.
/lgtm
/approve

**Pull request merged**